python/graphstorm/gconstruct/utils.py

```diff
@@ -304,7 +304,13 @@ def _merge_arrs(arrs, tensor_path):
     """
     assert isinstance(arrs, list)
     shape = _get_tot_shape(arrs)
-    dtype = arrs[0].dtype
+    # To get the output dtype by accessing the
+    # first element of the arrays (numpy array or HDFArray)
+    # Note: We use arrs[0][0] instead of arrs[0] because
+    #       arrs[0][0] is a transformed data with out_dtype
+    #       while arrs[0] can be a HDFArray and has not
+    #       been cast to out_dtype.
```

Comment:
does it mean that HDFArray has different dtype from the data type in array?

Reply:
HDFArray's dtype it is input dtype.

Reply:
i don't understand. what is input dtype?

Reply:
dtype of HDFArray._arr